File: C#/PBTKata/Post/PostalParcel.cs
using System;
using LanguageExt;
using static LanguageExt.Option<PBTKata.Post.PostalParcel>;

namespace PBTKata.Post
{
    public record PostalParcel
    {
        public double Weight { get; private init; }
        public static Option<PostalParcel> From(double weight) =>
            weight > 0 ? new PostalParcel { Weight = weight } : None;
    }

    public static class PostalParcelService
    {
        public const double MaxWeight = 20;
        public const double MinDeliveryCost = 1.99;
        public const double MaxDeliveryCost = 4.99;

        public static Option<double> CalculateDeliveryCosts(Option<PostalParcel> postalParcel) =>
            postalParcel.Map(p => p.Weight > MaxWeight ? MaxDeliveryCost : MinDeliveryCost);
    }
}

File: java/src/main/java/post/PostalParcel.java
package post;

import io.vavr.control.Option;

import static io.vavr.API.Some;
import static io.vavr.control.Option.none;

public record PostalParcel(double weight) {
    public static final double MAX_WEIGHT = 20d;
    public static final double MAX_DELIVERY_COSTS = 4.99;
    public static final double MIN_DELIVERY_COSTS = 1.99;

    public static Option<PostalParcel> from(double weight) {
        return weight > 0 ? Some(new PostalParcel(weight)) : none();
    }
}

